include/range/v3/view/take_while.hpp

```diff
@@ -98,6 +98,12 @@ namespace ranges
         {}
     };
 
+#if RANGES_CXX_DEDUCTION_GUIDES >= RANGES_CXX_DEDUCTION_GUIDES_17
+    CPP_template(typename Rng, typename Fun)(
+        requires ViewableRange<Rng>)
```

Comment:
Like `drop_while_view`, this needs `CopyConstructible<Fun>` (and `ViewableRange` could be removed).